MDAnalysis/mdanalysis
# moved analysis.x3dna to analysis.legacy.x3dna

Fixes #906 

Changes made in this Pull Request:
- establish the `MDAnalysis.analysis.legacy` module (following P2 in #743 and see also [Style Guide: Testing Analysis](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#tests-for-mdanalysisanalysis))
- demoted `x3dna` to legacy code


PR Checklist
------------
 - [x] Issue raised/referenced?

Details
-------

- created new MDAnalysis.analysis.legacy package
  - directory
  - docs
- x3dna is now considered legacy code (no/minimal testing, no maintenance)
- see #906 for reasons (mainly because the x3dna license does not allow us
  to just install it for testing on travis, or rather, licensing and access
  to x3dna is unknown/too complicated)
- closes #906
- see #743 on background for legacy code and also https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#tests-for-mdanalysisanalysis
- added stub with deprecation warning: until release 1.0, MDAnalysis.analysis.x3dna is still accessible
- added test case that the stub is there
